main.tf

```
@@ -40,61 +63,79 @@ resource "openstack_compute_instance_v2" "chef_zero" {
     }
 }
 
+resource "openstack_networking_port_v2" "source_server" {
+    name            = "source_server"
+    admin_state_up  = true
+    network_id      = data.openstack_networking_network_v2.public.id
+}
+
 resource "openstack_compute_instance_v2" "source" {
     name            = "source"
-    image_name      = "${var.centos_image}"
+    image_name      = var.centos_image
     flavor_name     = "m1.medium"
-    key_pair        = "${var.ssh_key_name}"
+    key_pair        = var.ssh_key_name
     security_groups = ["default"]
     connection {
         user = "centos"
+	host = openstack_networking_port_v2.source_server.all_fixed_ips.0
     }
     network {
-        uuid = "${data.openstack_networking_network_v2.network.id}"
+        port = openstack_networking_port_v2.source_server.id
     }
     network {
-        uuid        = "${openstack_networking_network_v2.mysql_network.id}"
-        fixed_ip_v4 = "192.168.60.11"
-    }
-    provisioner "chef" {
-        run_list        = ["role[mysql-vip]", "recipe[multi_node_test::source]"]
-        node_name       = "source"
-        secret_key      = "${file("test/integration/encrypted_data_bag_secret")}"
-        server_url      = "http://${openstack_compute_instance_v2.chef_zero.network.0.fixed_ip_v4}:8889"
-        recreate_client = true
-        user_name       = "fakeclient"
-        user_key        = "${file("test/chef-config/fakeclient.pem")}"
-        version         = "${var.chef_version}"
-        client_options  = ["chef_license 'accept'"]
+        port = openstack_networking_port_v2.source_port.id
+    }
+    provisioner "remote-exec" {
+        inline = ["echo online"]
     }
+    provisioner "local-exec" {
+        command = <<EOF
+            knife bootstrap -c test/chef-config/knife.rb \
+	    centos@${openstack_compute_instance_v2.source.network.0.fixed_ip_v4} \
+	    -y -N primary --secret-file test/integration/encrypted_data_bag_secret --sudo \
```

I'm running into an issue and it seems to be related to the fact that it's installing/using Chef 18 instead of 17. We're not quite ready for 18 so can you please add the following for each node?
```suggestion
	    -y -N primary --secret-file test/integration/encrypted_data_bag_secret --sudo --bootstrap-version 17 \
```